controller/readwrite.go

```diff
@@ -145,27 +145,28 @@ func (rw *ReadWrite) checkApply(ctx context.Context, u unit) (bool, error) {
 			taskName, err)
 	}
 
-	ev, err := event.NewEvent(taskName, &event.Config{
-		Providers: u.providers,
-		Services:  u.services,
-		Source:    u.source,
-	})
-	if err != nil {
-		return false, fmt.Errorf("error creating event for task %s: %s",
-			taskName, err)
-	}
-	defer func() {
-		ev.End(err)
-		log.Printf("[TRACE] (ctrl) adding event %s", ev.GoString())
-		err = rw.store.Add(*ev)
-	}()
-
 	// result.Complete is only `true` if the template has new data that has been
 	// completely fetched. Rendering a template for the first time may take several
 	// cycles to load all the dependencies asynchronously.
 	if result.Complete {
 		log.Printf("[DEBUG] (ctrl) change detected for task %s", taskName)
+
+		ev, err := event.NewEvent(taskName, &event.Config{
+			Providers: u.providers,
+			Services:  u.services,
+			Source:    u.source,
+		})
+		if err != nil {
+			return false, fmt.Errorf("error creating event for task %s: %s",
+				taskName, err)
+		}
 		ev.Start()
+		defer func() {
+			ev.End(err)
+			log.Printf("[TRACE] (ctrl) adding event %s", ev.GoString())
+			err = rw.store.Add(*ev)
```

Comment:
Since this error is set, was wondering if this error is supposed to go anywhere or supposed to be ignored? If the latter, probably best to omit the error return to make it more clear this error is ignored.

Reply:
Ack! Thank you for catching. I meant to log this error. I'll open a new PR to make this change (and others).